Release/4.0.0 (#6603)
open-formulieren/open-forms: docs/changelog-nl.rst

    found :ref:`here <changelog>`.

4.0.0 "Jongerius" (2026-08-31)
==============================

Open Formulieren 4.0.0 is een feature release met brekende wijzigingen.

.. epigraph::

   Jan Jongerius was een Nederlandse ondernemer, die garages had in meerdere Nederlandse steden en
   de grootste Ford-dealer was van Nederland. Villa Jongerius was gebouwd in Utrecht dicht bij de
   fabriekshallen - wat uiteindelijk de locatie werdt van het eerste Open Formulieren community
   evenement. De release-naam is een knipoog naar de gemeente Utrecht, die één van de bijdragers is

>    van deze release.

Reviewer: ```suggestion
   Jan Jongerius was een Nederlandse ondernemer die garages had in meerdere Nederlandse steden en
   de grootste Ford-dealer was van Nederland. Villa Jongerius was gebouwd in Utrecht dicht bij de
   bestaande werkplaatsen - wat uiteindelijk de locatie werdt van het eerste Open Formulieren 
   community-evenement. De releasenaam is een knipoog naar de gemeente Utrecht, die één van de bijdragers is van deze release.
```

Deze release bevat wijzigingen uit de alpha- en beta-versies en opgeloste bugs tot aan de
stabiele versie. Lees de release-opmerkingen aandachtig vóór het upgraden naar versie
4.0.0 en volg de onderstaande instructies.

Update-procedure
----------------

Om naar 4.0 te upgraden, let dan op:

* ⚠️ Zorg dat de huidige versie 3.5.5 of nieuwer is. We raden altijd de meest recente patch release

>   aan, op het moment van schrijven zijn dit 3.5.7.

Reviewer: ```suggestion
  aan, op het moment van schrijven is dit 3.5.7.
```

* ⚠️ Bekijk de :ref:`gedetailleerde release notes <installation_upgrade_400>` in de
  documentatie onder **Installation** > **Upgrade details to Open Forms 4.0.0** en
  bereid je hierop voor.

> * ⚠️ Zorg dat alle formulieren gebruik maken van de nieuwe logica evaluatie.

Reviewer: ```suggestion
* ⚠️ Zorg dat alle formulieren gebruik maken van de nieuwe logica-evaluatie.
```

> * ⚠️ Zorg dat (eigen) thema's zijn bijgewerkt en voorbereid voor 4.0.

Reviewer: ```suggestion
* ⚠️ Zorg dat (eigen) stijlen zijn bijgewerkt en voorbereid op 4.0.
```
as it's mentioned in the Dutch navigation menu in the admin

* ⚠️ Zorg dat verouderde ZGW URLs zijn gemigreerd als je de Objects API or ZGW APIs

>   registratieplugins gebruikt.

Reviewer: ```suggestion
* ⚠️ Zorg dat de verouderde ZGW-URLs gemigreerd zijn als je de Objecten-API- of ZGW-API's-
  registratieplugins gebruikt.
```

* ⚠️ Verifieer dat je bestaande formulieren nog werken met de bijgewerkte clear-on-hide

>   functionaliteit.

Reviewer: ```suggestion
* ⚠️ Verifieer dat je bestaande formulieren nog werken met de gewijzigde clear-on-hide
  functionaliteit.
```

brekende wijzigingen

> --------------------

Reviewer: ```suggestion
Brekende wijzigingen
--------------------
```

Bekijk de :ref:`4.0 release notes <installation_upgrade_400>` voor meer details.

* [:backend:`6164`] Verouderde code is verwijderd (bekijk de instructies in de documentatie onder

>   under **Installation** > **Upgrade details to Open Forms 4.0.0** voor alle details):

Reviewer: ```suggestion
* [:backend:`6164`] Verouderde code is verwijderd (bekijk de instructies in de documentatie onder
  **Installation** > **Upgrade details to Open Forms 4.0.0** voor alle details):
```

>   - De verouderde Camunda registratieplugin is verwijderd.

Reviewer: ```suggestion
  - De verouderde Camunda-registratieplugin is verwijderd.
```

  - De verouderde OIDC-callback-endpoints voor DigiD, eHerkenning en Organization login zijn

>     verwijderd.

Reviewer: ```suggestion
  - De verouderde OIDC-callback-endpoints voor DigiD-, eHerkenning- en Organisatielogin zijn
    verwijderd.
```

>   - De optie om de vorige formio.js-based Formulier renderer te gebruiken is verwijderd.

Reviewer: ```suggestion
  - De optie om de oude formio.js-gebaseerde renderer te gebruiken is verwijderd.
```

>   - De JCC SOAP afspraken plugin is verwijderd.

Reviewer: ```suggestion
  - De JCC SOAP-afsprakenplugin is verwijderd.
```

  - Ondersteuning voor Elastic APM is verwijderd.
  - De standaard ZGW APIs publieke referentie generatie is veranderd van zaaknummers naar

>     Open Formulieren-gegenereerde referenties.

Reviewer: ```suggestion
  - De standaardinstelling van de publieke referentiegeneratie in de  ZGW API's is veranderd van
    zaaknummers naar door Open Formulieren-gegenereerde referenties.
```

>   - Het ongebruikte ``authentication_success`` signaal is verwijderd.

Reviewer: ```suggestion
  - Het ongebruikte ``authentication_success``-signaal is verwijderd.
```

>   - Het v2 mede-ondertekenen ``cosign_date`` attribuut is nu gemarkeerd als verplicht.

Reviewer: ```suggestion
  - Het v2 mede-ondertekenen ``cosign_date``-attribuut is nu gemarkeerd als verplicht.
```

Reviewer: not sure if that properly works with the RST rendering though, so make sure to double check!

>   - UMD bundle wordt niet meer ondersteund.

Reviewer: ```suggestion
  - De UMD bundle wordt niet meer ondersteund.
```

  - Het Formio-vertalingen-endpoint is verwijderd.

>   - De standaard CSS-variabele voor ``backtotop-link`` zijn verwijderd.

Reviewer: ```suggestion
  - De standaard CSS-variabelen voor ``backtotop-link`` zijn verwijderd.
```

  - Haal Centraal BRP Personen bevragen 1.3 is niet meer ondersteund.

>   - De verouderde logica evaluatie is verwijderd.

Reviewer: ```suggestion
  - De verouderde logica-evaluatie is verwijderd.
```

>   - [:backend:`6311`] De Ogone Legacy betaalplugin is verwijderd.

Reviewer: ```suggestion
  - [:backend:`6311`] De Ogone Legacy-betaalplugin is verwijderd.
```

  - De verouderde detectie- en oplos-scripts zijn verwijderd.

Vanaf Open Formulieren 4.0 zal het KvK-vestigingsnummer bij voorkeur uit de
eHerkenning-logingegevens gehaald worden indien beschikbaar, in plaats van uit de
formuliergegevens (indien uitgevraagd).

  - [:backend:`6294`] Bij voorkeur wordt het KvK-vestigingsnummer nu uit de

>     eHerkenning-logingegevens gehaald, in plaats van uit de formuliergegevens (indien uitgevraagd).

Reviewer: ```suggestion
    eHerkenning-logingegevens gehaald, in plaats van uit de formuliergegevens (wanneer beide beschikbaar zijn).
```

  - [:backend:`4939`] De verouderde ``zaaktype`` en ``documenttype`` URL-based configuratieopties

>     voor de Objects API en ZGW APIs registratieplugins zijn verwijderd.

Reviewer: ```suggestion
  - [:backend:`4939`] De verouderde ``zaaktype``0 en ``documenttype``-URL-gebaseerde configuratieopties
    voor de Objecten-API- en ZGW API's-registratieplugins zijn verwijderd.
```

  - [:backend:`5769`] De programmeer-interfaces voor extenties zijn nu semi-private interfaces, wat
    betekent dat er brekende wijzigingen in kunnen gemaakt worden zonder dat dit leidt tot een

>     nieuwe major versie van Open Formulieren.

Reviewer: ```suggestion
  - [:backend:`5769`] De programmeerinterfaces voor extensies zijn nu semi-private interfaces, wat
    betekent dat er brekende wijzigingen in kunnen gemaakt worden zonder dat dit leidt tot een
    nieuwe major versie van Open Formulieren.
```

>   - De ``PERSIST_USER_DEFINED_VARIABLES_UPON_STEP_COMPLETION`` featureflag is verwijderd.

Reviewer: ```suggestion
  - De ``PERSIST_USER_DEFINED_VARIABLES_UPON_STEP_COMPLETION``-featureflag is verwijderd.
```

> * [:backend:`6275`] De functionaliteit van de "clear-on-hide" in de logica evaluatie is veranderd

Reviewer: ```suggestion
* [:backend:`6275`] De functionaliteit van de "Wissen als veld verborgen is" in de logica-evaluatie is gewijzigd
```

  - waardes die door het verbergen van het component zijn verwijderd, zijn nu volledig uitgesloten

>   bij de logica evaluatie. Hiervoor zouden deze componenten met de standaard of lege waardes

Reviewer: ```suggestion
  - waardes die door het verbergen van het component zijn verwijderd, zijn nu volledig afwezig
  bij de logica-evaluatie. Voorheen werd de standaard- of lege waarde van de componenten gebruikt.
```

  gebruikt worden in de logica evaluatie.
* [:backend:`6182`] De validatiefoutmeldingen en veldbeschrijving zijn verplaatst tussen het label
  en formulierinvoerveld voor verbeterde toegankelijkheid. De betreffende opmaak en CSS is

>   bijgewerkt, wat gevolgen kan hebben op eigen thema's.

Reviewer: ```suggestion
  bijgewerkt, wat gevolgen kan hebben op eigen stijlen.
```

Belangrijkste verbeteringen
---------------------------

> **💬 Enkele stap formulieren**

Reviewer: ```suggestion
**💬 Eénstapsformulieren**
```

Om het eenvoudiger te maken voor inwoners en bedrijven om digitaal contact op te nemen met

> overheidsinstanties, kan je nu "enkele stap" formulieren maken. Deze versimpelde vorm van

Reviewer: ```suggestion
overheidsinstanties, kan je nu "enkele stap"-formulieren maken. Deze versimpelde vorm van
```

> formulieren vereist geen gebruikers authenticatie, en kan direct ingevuld worden zonder dat een

Reviewer: ```suggestion
formulieren vraagt nooit om inloggen en kan direct ingevuld worden zonder dat eerst een
```

> formulierinzending gemaakt hoeft te worden. Het embedden van enkele stap formulieren kan het

Reviewer: ```suggestion
formulierinzending gestart moet worden. Het embedden van éénstapsformulieren kan het
```

eenvoudiger maken om te voldoen aan de
`WMEBV (Wet Modernisering Elektronisch Bestuurlijk Verkeer) <https://vng.nl/wmebv>`_ wetgeving.

> **♿️ Formulierveld layout aanpassingen**

Reviewer: ```suggestion
**♿️ Verbeterde formulierveldlayout**
```

De layout van formuliervelden is aangepast zodat veld beschrijvingen en validatiefoutmeldingen nu

> getoond worden tussen het veld label en invoerveld, in plaats van onder het invoerveld.

Reviewer: ```suggestion
De layout van formuliervelden is aangepast zodat beschrijvingen en foutmeldingen nu
getoond worden tussen het veldlabel en invoerveld, in plaats van onder het invoerveld.
```

This is een verzoek dat al langere tijd bekend is bij Open Formulieren. Dit trekt de UI en UX
gelijk met de aanbevelingen van het NL Design System, en biedt significante toegankelijkheids- en

> gebruikerservaring verbeteringen.

Reviewer: ```suggestion
Deze wens was al langere tijd bekend bij Open Formulieren maar was toen technisch lastig. Met deze
aanpassing brengen we de UI en UX in lijn met aanbevelingen uit NL Design System. Ze biedt significante
toegankelijkheids- en gebruikerservaringverbeteringen.
```

> **ℹ️ Hulp functie**

Reviewer: ```suggestion
**ℹ️ Hulpfunctie**
```

Een Hulp functie is toegevoegd om gebruikers ondersteuning te kunnen bieden bij het invullen van

> formulieren.

Reviewer: ```suggestion
De nieuwe hulpfunctie laat je toe om gebruikers ondersteuning te bieden tijdens het invullen van formulieren.
```

Een Hulp-uitlegpagina wordt getoond voordat het formulier wordt gestart om de ondersteuningsfunctie
te verklaren. Per formulierstap wordt een "hulp" knop getoond, die (als geklikt) de geconfigureerde

> hulp-instructies toont.

Reviewer: ```suggestion
Je kan een hulpuitlegpagina tonen voordat het formulier wordt gestart om de ondersteuningsfunctie
aan te wijzen. Wanneer hulpinstructies ingesteld zijn op het formulier, dan wordt op elke formulierstap
een hulpknop getoond waar de gebruiker op kan klikken om de instructies te zien.
```

**🏗️ Formulierbouwer**

> We hebben de formio.js-based formulierbouwer vervangen met onze eigen implementatie. Hiermee

Reviewer: ```suggestion
We hebben de formio.js-gebaseerde formulierbouwer vervangen met onze eigen implementatie. Hiermee
```

> hebben we enkele UI gerelateerde problemen kunnen oplossen, en konden afhankelijkheden met bekende

Reviewer: ```suggestion
hebben we enkele UI-gerelateerde problemen kunnen oplossen, en konden softwarebibliotheken met bekende
```

veiligheidsproblemen bijgewerkt of verwijderd worden.

Dit betekent ook dat toekomstige functionaliteiten, wensen en verbeteringen gedaan kunnen worden

> zonder externe aanpassingen.

Reviewer: ```suggestion
Dit betekent ook dat toekomstige functionaliteiten, wensen en verbeteringen gedaan kunnen worden
zonder externe afhankelijkheden.
```

Nieuwe functies
---------------

* [:backend:`5932`] Een 'gebruik email voor bevestigingsemail' optie is toegevoegd aan het

>   ``customerProfile``-formuliercomponent.

Reviewer: ```suggestion
* [:backend:`5932`] Een 'gebruik email voor bevestigingsmail' optie is toegevoegd aan het
  ``customerProfile``-formuliercomponent.
```
bevestigingsmail or bevestigings-e-mail, pick one 😛

> * [:backend:`5764`] Enkele stap formulieren:

Reviewer: ```suggestion
* [:backend:`5764`] Eénstapsformulieren:
```

  - Het toevoegen van 'enkele stap' formulieren, die geen login- of startpagina bevatten. Deze

>     formulieren kunnen direct ingevuld en ingezonden worden.

Reviewer: ```suggestion
  - Het toevoegen van 'enkele stap'-formulieren, die geen login- of startpagina bevatten. Deze
    formulieren kunnen direct ingevuld en ingezonden worden.
```

  - [:backend:`6098`] Het formulier-attribuut ``is_appointment`` is vervangen voor het ``type``
    formulier-attribuut. Het ``type`` attribuut biedt enkele opties:
    ``Standaard`` (de standaard waarde), ``Afspraak`` en ``Enkele stap``.
  - [:backend:`6106`] De formulier designer UI is bijgewerkt voor het nieuwe ``Enkele stap``

>     formuliertype.

Reviewer: ```suggestion
  - [:backend:`6106`] De formulierdesigner-UI is bijgewerkt voor het nieuwe ``Enkele stap``
    formuliertype.
```

  - [:backend:`6211`] Je kan formulieren nu filteren op type.
  - [:backend:`6350`] Een optie is toegevoegd om de formuliertitel te verbergen.
  - [:backend:`6398`] Een vast ``form_url`` variabele is toegevoegd. Dit bevat metadata over waar

>     een formulierinzending is begonnen.

Reviewer: ```suggestion
  - [:backend:`6398`] Er is nu een vaste variabele ``form_url``. Deze bevat metadata over waar
    een formulierinzending is begonnen.
```

  - [:backend:`6399`] Enkele stap formulieren kunnen variabele waardes veranderen en DMN evaluaties

>     uitvoeren met logicaregels.

Reviewer: ```suggestion
  - [:backend:`6399`] "Enkele stap"-formulieren kunnen variabelewaardes veranderen en DMN-evaluatie
    uitvoeren met logicaregels.
```

* Enkele tijdelijke oplossingen zijn toegevoegd voor StUF-ZDS wanneer Onegov zaaksysteem wordt

>   gebruikt:

Reviewer: ```suggestion
* Er zijn workarounds toegevoegd in de StUF-ZDS-registratieplugin voor het Onegov-zaaksysteem:
```

>   - [:backend:`6230`] Biedt de inzending ``completed_on`` timestamp aan als registratievariabele.

Reviewer: ```suggestion
  - [:backend:`6230`] De ``completed_on``-registratievariabele van de inzending is nu beschikbaar.
```

  - [:backend:`5951`] Ondersteuning toegevoegd voor ``extraElementen`` mapping-configuratie op het
    initiator-niveau.
  - [:backend:`6242`] Optie toegevoegd om ``extraElement`` lijst waardes als komma-gescheiden tekst

>     te gebruiken.

Reviewer: ```suggestion
  - [:backend:`6242`] Optie toegevoegd om ``extraElement`` lijstwaardes als komma-gescheiden waarde
    aan te bieden.
```

* [:backend:`5959`] Ondersteuning toegevoegd voor ``variabelen`` op het nieuwe (v3)

>   formulier-endpoint.

Reviewer: ```suggestion
* [:backend:`5959`] Ondersteuning toegevoegd voor ``variables`` op het nieuwe (v3)
  formulier-endpoint.
```

> * Verschillende aanpassingen aan de ZGW APIs registratieplugin:

Reviewer: ```suggestion
* Verschillende aanpassingen aan de ZGW-API's-registratieplugin:
```

  - [:backend:`5676`] Configuratievelden voor ``Omschrijving`` en ``Toelichting`` zijn toegevoegd

>     aan de ZGW APIs registratieplugin.

Reviewer: ```suggestion
    aan de ZGW-API's-registratieplugin.
```

>   - [:backend:`4939`] Een upgrade check is toegvoegd om het gebruik van verouderde ZGW APIs URLs te

Reviewer: ```suggestion
  - [:backend:`4939`] Een upgrade check is toegvoegd om het gebruik van verouderde ZGW-API's-URL-referenties te
```

    dedecteren.

>   - [:backend:`6217`] In de ZGW APIs registratieplugin kan je nu een JSON-document met de

Reviewer: ```suggestion
  - [:backend:`6217`] In de ZGW-API's-registratieplugin kan je nu een JSON-document met de
```
... you get the gist I think 😬 any additional occurrence should be updated

    inzendingdetails toevoegen aan de zaak en selecteren welk bestandsformaten gebruikt zullen

>     worden (JSON en/of PDF).

Reviewer: ```suggestion
    inzendingdetails toevoegen aan de zaak en selecteren welk bestandsformaten toegevoegd moeten
    worden (JSON en/of PDF).
```

  - [:backend:`5984`] Een ``contactpersoonRol`` attribuut toegevoegd voor wanneer rol-objecten via
    de ZGW APIs registratieplugin worden aangemaakt.

>   - [:backend:`6343`] Je kan nu 'overige' zaak objecten toevoegen in de ZGW APIs registratieplugin.

Reviewer: ```suggestion
  - [:backend:`6343`] Je kan nu 'overige' zaakobjecten toevoegen in de ZGW-API's-registratieplugin.
```

* [:backend:`6274`] Een script is toevoegd om brekende aanpassingen te identificeren in verband met

>   de clear-on-hide aanpassingen.

Reviewer: ```suggestion
* [:backend:`6274`] Een script is toevoegd om brekende aanpassingen te identificeren door de
  "wissen als veld verborgen is"-aanpassingen.
```

* [:backend:`6391`] Een knop is toegevoegd om simpele logicaregels op te zetten naar geavanceerde

>   logicaregels.

Reviewer: ```suggestion
* [:backend:`6391`] Een knop is toegevoegd om simpele logicaregels om te zetten naar geavanceerde
  logicaregels.
```

* [:backend:`6351`] De beschikbare formulier-componenenten in de formulierbouwer worden nu

>   aangepast wanneer het formuliertype veranderd.

Reviewer: ```suggestion
* [:backend:`6351`] De beschikbare formulier-omponenenten in de formulierbouwer zijn nu afhankelijk van
  het ingestelde formuliertype.
```

* [:backend:`6281`] Je kan nu de document registratieopties specificeren voor bestandsupload

>   componenten per registratieplugin die wordt gebruikt in het formulier:

Reviewer: ```suggestion
* [:backend:`6281`] Je kan nu de documentregistratieopties van bestandsuploadcomponenten per
  registratieplugin in het formulier instellen:
```

  - Registratieconfiguratie toegevoegd voor Objects API en ZGW APIs plugins op de bestandsupload

>     component variabeleconfiguratie.

Reviewer: ```suggestion
  - Registratieconfiguratie toegevoegd voor Objecten-API- en ZGW-API's-plugins op de
    bestandsuploadcomponentvariabeleconfiguratie.
```
that's a nice word lol

  - De registratie-tab op de bestandsupload componentconfiguratie is verwijderd.
  - Een migratie is toegevoegd om automatisch de registratieconfiguratie op het bestandsupload
    component te verplaatsen naar de betreffende variabeleregistratieconfiguratie.
  - Documentatie toegevoegd over het gebruik van de nieuwe bestandsupload component

>     variabeleregistratieconfiguratie.

Reviewer: ```suggestion
  - De registratietab op de bestandsuploadcomponentconfiguratie is verwijderd.
  - Een migratie is toegevoegd om automatisch de registratieconfiguratie op de bestandsuploadcomponent
    te verplaatsen naar de betreffende variabeleregistratieconfiguratie.
  - Documentatie toegevoegd over het gebruik van de nieuwe
    bestandsuploadomponentvariabeleregistratieconfiguratie.
```
_de_ component, always :)

* [:backend:`6336`] Wanneer de StUF-ZDS of ZGW APIs registratieplugins worden gebruikt, is er nu
  een ``address`` registratieattribuut beschikbaar voor ``addressNL`` componenten.

> * Enkele importeer conversies zijn toegevoegd voor verouderende formulier exports:

Reviewer: ```suggestion
* Enkele importeerconversies zijn toegevoegd voor verouderende formulierexports:
```

  - [:backend:`6281`] Conversie voor bestandsupload component registratieopties.
  - Een migratie en conversie voor het oplossen van foutieve ``Datum`` en ``Datum en tijd``

>     component minimum/maximum datum configuraties.

Reviewer: ```suggestion
  - Een migratie en conversie voor het oplossen van foutieve ``date``- en ``dateTime``-component minimum/maximum datumconfiguraties.
```

  - [:backend:`6349`] Conversie voor afspraakformulieren met betrekking tot de formuliertype

>     aanpassing.

Reviewer: ```suggestion
  - [:backend:`6349`] Conversie voor afspraakformulieren met betrekking tot de formuliertype-aanpassing.
```

> * [:backend:`6380`] De ``substr`` JSON logica operator is nu beschikbaar in de logicaregels.

Reviewer: ```suggestion
* [:backend:`6380`] De ``substr`` JsonLogic-operator is nu beschikbaar in de logicaregels.
```

> * [:backend:`5377`] De meeste "Haal Centraal BRP Personen bevragen" attributen zijn nu beschikbaar

Reviewer: ```suggestion
* [:backend:`5377`] De meeste "Haal Centraal BRP Personen bevragen"-attributen zijn nu beschikbaar
```
